R/import_ebird.R

```diff
@@ -83,5 +150,56 @@ ebird_col_type <- function(col_names) {
   setNames(col_types, col_names)
 }
 
-# Download URLs, eg https://download.ebird.org/ebd/prepackaged/ebd_relJul-2021.tar
-# Note, may take > 24hrs to complete
+parse_ebd_filename <- function(tarfile) {
+  stopifnot(is.character(tarfile), length(tarfile) == 1, file.exists(tarfile))
+  
+  f <- basename(tarfile)
+  # checks for validity of filename
+  if (!grepl("\\.tar$", f)) {
+    stop("The provided file does not appear to be a tar archive. The file ",
+         "extension should be .tar.")
+  }
+  if (grepl("ebd_sampling_rel[A-Z]{1}[a-z]{2}-[0-9]{4}\\.tar$", f)) {
+    dataset <- "checklists"
+    subset <- NA_character_
+  } else if (grepl("ebd_rel[A-Z]{1}[a-z]{2}-[0-9]{4}\\.tar$", f)) {
+    dataset <- "observations"
+    subset <- NA_character_
+  } else if (grepl("ebd_[-_A-Za-z0-9]+_rel[A-Z]{1}[a-z]{2}-[0-9]{4}\\.tar$", f)) {
+    dataset <- "observations"
+    subset <- sub("ebd_([-_A-Za-z0-9]+)_rel[A-Z]{1}[a-z]{2}-[0-9]{4}\\.tar", 
+                  "\\1", f)
+  } else {
+    stop("The provided tar filename does not appear to contain eBird data. ", 
+         "The expected format is, e.g., ebd_relJul-2021.tar.")
+  }
+  
+  # parse date from filename
+  rawdate <- sub("ebd[-_A-Za-z0-9]*_rel([A-Z]{1}[a-z]{2}-[0-9]{4})\\.tar", 
+                 "\\1", f)
+  date <- strsplit(rawdate, "-")[[1]]
+  date[1] <- match(date[1], month.abb)
+  date <- paste(date[2], date[1], "1", sep = "-")
+  date <- as.Date(date, format = "%Y-%m-%d")
+  if (is.na(date)) {
+    stop("Month and year could not be parsed from filename: ", rawdate)
+  }
+  version = format(date, "%Y-%m")
+
+  msg <- sprintf("Importing %s data from the %s eBird Basic Dataset: %s",
```

Comment:
A message is printed specifying what is above to be imported. Any reason to optionally turn this off with a `verbose` parameter? I think it's wise tell people what they're about to do here.

Reply:
I think this makes sense.  I don't think a `verbose` parameter is needed; users can rely on standard R msg suppression if they really need it.  

Speaking of messages, obviously one thing that would be really nice is some kind of progress bar or eta for the long running tasks, but I think that's probably pretty difficult here (unless something very clever comes out upstream) so we don't worry about that for now.